service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractStatusService.java

```diff
@@ -111,17 +119,24 @@ public EhrExtractStatus updateEhrExtractStatusAccessDocument(GetGpcDocumentTaskD
         return ehrExtractStatus;
     }
 
-    public void updateEhrExtractStatusCore(SendEhrExtractCoreTaskDefinition sendEhrExtractCoreTaskDefinition, Instant requestSentAt) {
+    public EhrExtractStatus updateEhrExtractStatusCore(SendEhrExtractCoreTaskDefinition sendEhrExtractCoreTaskDefinition,
+        Instant requestSentAt) {
         Query query = createQueryForConversationId(sendEhrExtractCoreTaskDefinition.getConversationId());
 
         Update update = createUpdateWithUpdatedAt();
         update.set(EXTRACT_CORE_SENT_AT_PATH, requestSentAt);
         update.set(EXTRACT_CORE_TASK_ID_PATH, sendEhrExtractCoreTaskDefinition.getTaskId());
-        UpdateResult updateResult = mongoTemplate.updateFirst(query, update, EhrExtractStatus.class);
+        FindAndModifyOptions returningUpdatedRecordOption = getReturningUpdatedRecordOption();
 
-        if (updateResult.getModifiedCount() != 1) {
+        EhrExtractStatus ehrExtractStatus =
+            mongoTemplate.findAndModify(query, update, returningUpdatedRecordOption, EhrExtractStatus.class);
+
+        if (ehrExtractStatus == null) {
             throw new EhrExtractException("EHR Extract Status was not updated with Extract Core Message.");
         }
+
+        LOGGER.info("Database successfully updated with EHRContinue");
```

Comment:
This log message does not describe what is actually happening here

```suggestion
        LOGGER.info("Database successfully updated after sending EhrExtract");
```